File: plugins/scandit-sdk/evals/sparkscan-net-android/fixtures/IntegratedActivity.cs
using Android.OS;
using AndroidX.AppCompat.App;

using Scandit.DataCapture.Barcode.Data;
using Scandit.DataCapture.Barcode.Spark.Capture;
using Scandit.DataCapture.Barcode.Spark.Feedback;
using Scandit.DataCapture.Barcode.Spark.UI;
using Scandit.DataCapture.Barcode.Spark.UI.Platform.Android;
using Scandit.DataCapture.Core.Capture;

namespace MyApp;

[Activity(MainLauncher = true, Label = "@string/app_name")]
public class MainActivity : AppCompatActivity, ISparkScanFeedbackDelegate
{
    public const string SCANDIT_LICENSE_KEY = "-- ENTER YOUR SCANDIT LICENSE KEY HERE --";

    private DataCaptureContext dataCaptureContext = null!;
    private SparkScan sparkScan = null!;
    private SparkScanView sparkScanView = null!;

    private SparkScanBarcodeSuccessFeedback successFeedback = null!;
    private SparkScanBarcodeErrorFeedback errorFeedback = null!;

    protected override void OnCreate(Bundle? savedInstanceState)
    {
        base.OnCreate(savedInstanceState);
        this.SetContentView(Resource.Layout.activity_main);
        this.Initialize();
    }

    protected override void OnPause()
    {
        base.OnPause();
        this.sparkScanView.OnPause();
    }

    protected override void OnResume()
    {
        base.OnResume();
        this.sparkScanView.OnResume();
    }

    private void Initialize()
    {
        this.dataCaptureContext = DataCaptureContext.ForLicenseKey(SCANDIT_LICENSE_KEY);

        SparkScanSettings settings = new SparkScanSettings();
        settings.EnableSymbology(Symbology.Ean13Upca, true);
        settings.EnableSymbology(Symbology.Code128, true);

        this.sparkScan = new SparkScan(settings);
        this.sparkScan.BarcodeScanned += this.OnBarcodeScanned;

        SparkScanCoordinatorLayout container =
            this.FindViewById<SparkScanCoordinatorLayout>(Resource.Id.spark_scan_coordinator)!;
        SparkScanViewSettings viewSettings = new SparkScanViewSettings();

        this.sparkScanView = SparkScanView.Create(
            parentView: container,
            context: this.dataCaptureContext,
            sparkScan: this.sparkScan,
            settings: viewSettings);

        this.successFeedback = new SparkScanBarcodeSuccessFeedback();
        this.errorFeedback = new SparkScanBarcodeErrorFeedback(
            message: "Wrong barcode",
            resumeCapturingDelay: TimeSpan.FromSeconds(60));
        this.sparkScanView.Feedback = this;
    }

    private void OnBarcodeScanned(object? sender, SparkScanEventArgs args)
    {
        var barcode = args.Session.NewlyRecognizedBarcode;
        if (barcode == null) return;

        this.RunOnUiThread(() => { });
    }

    SparkScanBarcodeFeedback ISparkScanFeedbackDelegate.GetFeedbackForBarcode(Barcode barcode) =>
        IsBarcodeValid(barcode) ? this.successFeedback : this.errorFeedback;

    private static bool IsBarcodeValid(Barcode barcode) => barcode.Data != "123456789";
}

File: plugins/scandit-sdk/evals/sparkscan-android/fixtures/IntegratedActivity.kt
import android.os.Bundle
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import com.scandit.datacapture.barcode.data.Symbology
import com.scandit.datacapture.barcode.spark.capture.SparkScan
import com.scandit.datacapture.barcode.spark.capture.SparkScanListener
import com.scandit.datacapture.barcode.spark.capture.SparkScanSession
import com.scandit.datacapture.barcode.spark.capture.SparkScanSettings
import com.scandit.datacapture.barcode.spark.ui.SparkScanView
import com.scandit.datacapture.barcode.spark.ui.SparkScanViewSettings
import com.scandit.datacapture.core.capture.DataCaptureContext
import com.scandit.datacapture.core.data.FrameData

class MainActivity : AppCompatActivity(), SparkScanListener {

    private lateinit var dataCaptureContext: DataCaptureContext
    private lateinit var sparkScan: SparkScan
    private lateinit var sparkScanView: SparkScanView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setupScanning()
    }

    override fun onResume() {
        super.onResume()
        sparkScanView.onResume()
    }

    override fun onPause() {
        sparkScanView.onPause()
        super.onPause()
    }

    private fun setupScanning() {
        dataCaptureContext = DataCaptureContext.forLicenseKey("-- ENTER YOUR SCANDIT LICENSE KEY HERE --")

        val settings = SparkScanSettings().apply {
            enableSymbologies(setOf(Symbology.EAN13_UPCA, Symbology.CODE128))
        }
        sparkScan = SparkScan(settings)
        sparkScan.addListener(this)

        val viewSettings = SparkScanViewSettings()
        sparkScanView = SparkScanView.newInstance(
            findViewById<ViewGroup>(android.R.id.content),
            dataCaptureContext,
            sparkScan,
            viewSettings
        )
    }

    override fun onBarcodeScanned(
        sparkScan: SparkScan,
        session: SparkScanSession,
        data: FrameData?,
    ) {
        val barcode = session.newlyRecognizedBarcode ?: return
        runOnUiThread {
            println("Scanned: ${barcode.data}")
        }
    }
}
